### PR TITLE
Improve TypeUuid's derive macro error messages

### PR DESCRIPTION
**Pull request by tguichaoua**
# Objective

- Better error message
- More idiomatic code

## Solution

Refactorize `TypeUuid` macros to use `syn::Result` instead of panic.

#### Before
```
error: proc-macro derive panicked
 --> src\main.rs:1:10
  |
1 | #[derive(TypeUuid)]
  |          ^^^^^^^^
  |
  = help: message: No `#[uuid = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"` attribute found.
```

#### After
```
error: No `#[uuid = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"]` attribute found.
 --> src\main.rs:3:10
  |
3 | #[derive(TypeUuid)]
  |          ^^^^^^^^
  |
  = note: this error originates in the derive macro `TypeUuid` (in Nightly builds, run with -Z macro-backtrace for more info)
```

#### Before

```
error: proc-macro derive panicked
 --> src\main.rs:3:10
  |
3 | #[derive(TypeUuid)]
  |          ^^^^^^^^
  |
  = help: message: `uuid` attribute must take the form `#[uuid = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"`.
```

#### After

```
error: `uuid` attribute must take the form `#[uuid = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"]`.
 --> src\main.rs:4:1
  |
4 | #[uuid = 42]
  | ^^^^^^^^^^^^
```

#### Before
```
error: proc-macro derive panicked
 --> src\main.rs:3:10
  |
3 | #[derive(TypeUuid)]
  |          ^^^^^^^^
  |
  = help: message: Value specified to `#[uuid]` attribute is not a valid UUID.: Error(SimpleLength { len: 3 })
```

#### After

```
error: Invalid UUID: invalid length: expected length 32 for simple format, found 3
 --> src\main.rs:4:10
  |
4 | #[uuid = "000"]
  |          ^^^^^
```

#### Before
image

#### After
image

---

## Changelog

- `#[derive(TypeUuid)]` provide better error messages.

### CODE REVIEW

File: crates/bevy_reflect/bevy_reflect_derive/src/type_uuid.rs [hunk 1]
```diff
@@ -90,10 +89,10 @@ pub(crate) struct TypeUuidDef {
 }
 
 impl Parse for TypeUuidDef {
-    fn parse(input: ParseStream) -> Result<Self> {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
         let type_ident = input.parse::<Ident>()?;
         let generics = input.parse::<Generics>()?;
-        input.parse::<Comma>()?;
+        input.parse::<Token![,]>()?;
```
Comment (mockersf) [inline crates/bevy_reflect/bevy_reflect_derive/src/type_uuid.rs:95]: Could you revert this change?

**Review by mockersf — APPROVED** [review]

**Review by james7132 — APPROVED** [review]
LGTM. It'd be nice to have a UX test (like `bevy_ecs_compile_fail_tests`) for TypeUuid, but I won't block on that.
